main.go

```diff
@@ -3,13 +3,38 @@ package main
 import (
 	"log"
 	"net/http"
+	"net/http/httputil"
+	"net/url"
 	"observability-remote-write-proxy/pkg/remotewrite"
 )
 
+const (
+	prefixHeader = "X-Forwarded-Prefix"
+)
+
 func main() {
-	handler := http.HandlerFunc(handleRequestSuccess)
-	http.Handle("/", handler)
-	err := http.ListenAndServe(":8080", nil)
+	upstreamUrl, err := url.Parse("http://localhost:9091/api/v1/write")
+	if err != nil {
+		panic(err)
+	}
+
+	proxy := httputil.ReverseProxy{
+		Director: func(request *http.Request) {
+			request.URL.Scheme = upstreamUrl.Scheme
+			request.Host = upstreamUrl.Host
+			request.URL.Host = upstreamUrl.Host
+			request.URL.Path = upstreamUrl.Path
+			request.Header.Add(prefixHeader, "/")
+		},
+		ErrorHandler: func(writer http.ResponseWriter, request *http.Request, err error) {
+			log.Println(err)
+		},
+	}
+
+	proxy.Transport = http.DefaultTransport
+	http.Handle("/", http.HandlerFunc(handleRequestSuccess))
+	http.Handle("/", &proxy)
```

Comment:
What happens if the request is invalid? Are we still going to proxy the request?

Reply:
I don't think there is a need to proxy it if it is invalid, will add a check. What do you think @pb82 ?